sdk/core/azure-core/azure/core/tracing/abstract_span.py

```diff
@@ -10,11 +10,19 @@
 except AttributeError:  # Python 2.7, abc exists, but not ABC
     ABC = abc.ABCMeta("ABC", (object,), {"__slots__": ()})  # type: ignore
 
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Optional
+
 
 class AbstractSpan(ABC):
     @abstractmethod
     def __init__(self, span=None, name=None):
-        # type: (Any, str) -> None
+        # type: (Optional[Any], Optional[str]) -> None
```

Comment:
I would consider having a protocol for span.

Reply:
I chose abstract class because I like the idea that we could check if the user given plugin is a subclass of AbstractSpan. Are we ok with just using whatever the user gives instead of doing that check?

Reply:
is protocol now.